packages/astro/src/@types/astro.ts

```diff
@@ -2408,6 +2408,131 @@ export interface AstroUserConfig {
 		 * For a complete overview and the full API reference, see [the Content Layer API RFC](https://github.com/withastro/roadmap/blob/content-layer/proposals/0047-content-layer.md) and [share your feedback](https://github.com/withastro/roadmap/pull/982).
 		 */
 		contentLayer?: boolean;
+
+		/**
+		 * @docs
+		 * @name experimental.tsconfig
+		 * @type {object}
+		 * @default `undefined`
+		 * @version 4.15.0
+		 * @description
+		 *
+		 * Enables a streamlined way of configuring TypeScript in your project. Enjoy an improved DX when setting up Astro for Intellisense, type-checking, and more.
+		 *
+		 * This option removes the need for a separate `src/env.d.ts` file to provide Vite's client types to your project. Instead, `/.astro/tsconfig.json` will be autogenerated with linked types for you. Notably, this fixes an existing issue with `astro check` false positives by excluding `outDir` from your checked files by default.
```

Comment:
```suggestion
		 * This option removes the need for a separate `src/env.d.ts` file to provide Vite's client types to your project. Instead, `/.astro/tsconfig.json` will be autogenerated with linked types for you. Notably, this fixes an existing issue with `astro check` false positives by excluding your build from your checked files by default.
```
 Trying to make this as little jargony as I can (since we do explain the config item later). Can we say this?